libraries/rush-lib/src/logic/CredentialCache.ts

```diff
@@ -67,8 +70,18 @@ export class CredentialCache /* implements IDisposable */ {
   }
 
   public static async initializeAsync(options: ICredentialCacheOptions): Promise<CredentialCache> {
-    const rushUserFolderPath: string = RushUserConfiguration.getRushUserFolderPath();
-    const cacheFilePath: string = `${rushUserFolderPath}/${CACHE_FILENAME}`;
+    let cacheDirectory: string;
+    let cacheFileName: string;
+    if (options.cacheFilePath) {
+      cacheDirectory = path.dirname(options.cacheFilePath);
+      // Ensure .json extension
+      cacheFileName = path.basename(options.cacheFilePath, CACHE_FILE_EXTENSION) + CACHE_FILE_EXTENSION;
```

Comment:
The `.json` extension doesn't actually matter. Someone can pass something in without and thins will work fine.